packages/component-submission/client/components/WizardStep.js

```diff
@@ -48,7 +47,8 @@ class WizardStep extends React.Component {
         // ensure each page gets a new form instance otherwise all fields are touched
         key={title}
         onSubmit={values => {
-          handleButtonClick(values).then(() => history.push(nextUrl))
+          handleButtonClick({ ...values, lastStepVisited: nextUrl })
```

Comment:
This was my own change but will moving the `history.push` out of the resolved promise cause any issues when the form does need to save when navigating? will we lose the last 5 seconds of data?